lnwallet/channel.go

```diff
@@ -4840,31 +4858,86 @@ func (lc *LightningChannel) ReceiveNewCommitment(commitSigs *CommitSigs) error {
 	cancelChan := make(chan struct{})
 	verifyResps := lc.sigPool.SubmitVerifyBatch(verifyJobs, cancelChan)
 
+	localCommitTx := localCommitmentView.txn
+	multiSigScript := lc.signDesc.WitnessScript
+	prevFetcher := txscript.NewCannedPrevOutputFetcher(
+		multiSigScript, int64(lc.channelState.Capacity),
+	)
+	hashCache := txscript.NewTxSigHashes(localCommitTx, prevFetcher)
+
 	// While the HTLC verification jobs are proceeding asynchronously,
 	// we'll ensure that the newly constructed commitment state has a valid
 	// signature.
-	verifyKey := lc.channelState.RemoteChanCfg.MultiSigKey.PubKey
+	//
+	// To do that we'll, construct the sighash of the commitment
+	// transaction corresponding to this newly proposed state update.  If
+	// this is a taproot channel, then in order to validate the sighash,
+	// we'll need to call into the relevant tapscript methods.
+	if lc.channelState.ChanType.IsTaproot() {
+		localSession := lc.musigSessions.LocalSession
+
+		// As we want to ensure we never write nonces to disk, we'll
+		// use the shachain state to generate a nonce for our next
+		// local state. Similar to generateRevocation, we do height + 2
+		// (next height + 1) here, as this is for the _next_ local
+		// state, and we're about to accept height + 1.
+		localCtrNonce := WithLocalCounterNonce(
```

Comment:
Also looks like it's generating a new nonce here. Since we are receiving a `commitment_signed` here, we should use the nonce we sent before in `channel_ready`, `revoke_and_ack` or `channel_reestablish`, but not generating a new one here?

Reply:
Yep it's a functional option used to give the session state a canned nonce.